### ISSUE TITLE
Question: Why does this library require Node >= 16.x.x?

### ISSUE THREAD

**Issue by mattseddon** [extremeheat/JSPyBridge#32]
Tl;Dr - why does this library need `Node >= 16.x.x`?.

Hi, 

I am currently working on a VS Code extension for [DVC](https://dvc.org/). To give you some context DVC is a CLI for (not just) Data Version Control in Machine Learning Projects. The CLI is written in Python and the extension is written in Typescript then bundled and run inside of VS Code (which is an Electron app). I am currently looking to [improve the integration](https://github.com/iterative/vscode-dvc/issues/967) between the CLI and the extension. 

At the moment we use [execa](https://github.com/sindresorhus/execa) to create `child_process`es which run the CLI as a black box and return JSON. We then parse that JSON and feed it through the extension. 

Not being able to access the internals of the Python project has started to become an issue so we have been looking for alternatives. This looks like a solid library but unfortunately, the VS Code Electron app runs on Node 14.17.0 and as far as I am aware there are no plans to change this in the short to medium term.

What would need to be done to make this library compatible with an earlier version of Node? Is that even possible?

Thanks,
Matt

**Comment by extremeheat** [comment]
in #33:
> I believe the main issue with node 14 was with the javascript pip package where the event emitter handling [code changed](https://github.com/nodejs/node/pull/35931) in node 15/16 causing the behavior on the Python side to be different. It was fixed with a patch [here](https://github.com/extremeheat/JSPyBridge/blob/master/src/javascript/__init__.py#L78). I don't think the issue is present on the pythonia package as we don't do event emitter handling there, so yeah doesn't seem like there's any other blocking issue for node 16.

so once we merge that we can probably do a release if everything looks good

**Comment by mattseddon** [comment]
Thank you both.

**Comment by extremeheat** [comment]
Fixed in #33 and released as 0.2.13

**Comment by mattseddon** [comment]
Thanks, I'll come back and let you know how everything goes.
